src/app/features/vendor/vendor-creation/vendor-creation.component.spec.ts

```diff
@@ -0,0 +1,266 @@
+    const getServicesSpy = jest
+      .spyOn(vendorService, 'getServices')
+      .mockImplementation(() => of(getService()));
+    const getVendorByIdSpy = jest
+      .spyOn(vendorService, 'getVendorById')
+      .mockImplementation(() => of(getVendorData()));
+    const addVendorSpy = jest
+      .spyOn(vendorService, 'addVendor')
+      .mockImplementation(() => of(returnResponse()));
+    const updateVendorSpy = jest
+      .spyOn(vendorService, 'updateVendor')
+      .mockImplementation(() => of(returnResponse()));
+    const routerSpy = jest
+      .spyOn(router, 'navigate')
+      .mockImplementation(() => Promise.resolve(true));
+    const messageServiceSpy = jest
+      .spyOn(messageService, 'add')
+      .mockImplementation(() => {});
+    return {
+      getMarketsSpy,
+      getServicesSpy,
+      getVendorByIdSpy,
+      addVendorSpy,
+      updateVendorSpy,
+      routerSpy,
+      messageServiceSpy,
+    };
+  };
+  let urlValue = { url: 'vendor/creation', navigate: jest.fn() };
+
+  it('should create', () => {
+    const { component } = setup();
+    expect(component).toBeTruthy();
+  });
+
+  it('should create vendor form on component initialization', () => {
+    const { component } = setup();
+    component.ngOnInit();
+    expect(component.addVendorForm).toBeDefined();
+  });
+
+  it('should fetch vendor data and bind to form when editing', () => {
+    const { component } = setup();
+    urlValue.url = 'vendor/edit/1';
+    component.ngOnInit();
+    component.fetchVendorData();
+    expect(component.addVendorForm.get('vendorName')?.value).toBe('Vendor 1');
+  });
+
+  it('should update the vendor data during update form submission', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = true;
+    component.selectedVendorId = 1;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.updateVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should handle error during update the vendor data', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = true;
+    component.selectedVendorId = 1;
+    component.addVendorForm = getVendorServiceFormMock();
+    const errorResponse = {
+      error: {
+        errors: {
+          error: [
+            {
+              message: '',
+            },
+          ],
+        },
+        message: 'Error occurred',
+      },
+    };
+    spyObj.updateVendorSpy.mockReturnValue(throwError(() => errorResponse));
+    component.submitVendor();
+    expect(spyObj.updateVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should add the vendor data during  form submission', () => {
+    const { component, spyObj } = setup();
+    component.isEdit = false;
+    component.addVendorForm = getVendorServiceFormMock();
+    component.submitVendor();
+    expect(spyObj.addVendorSpy).toHaveBeenCalled();
+  });
+
+  it('should handle validation error during vendor form submission', () => {
```

Comment:
Description should not be technical. 
eg: should show error message when there is an error occurs during vendor creation / update